netam/model_factory.py

```diff
@@ -0,0 +1,395 @@
+"""Model factory functions for creating netam models.
+
+This module provides factory functions to create netam selection models from
+configuration dictionaries and files. Supports all model types including:
+- SingleValueBinarySelectionModel
+- ParentIndependentBinarySelectionModel
+- TransformerBinarySelectionModelLinAct
+- TransformerBinarySelectionModelWiggleAct
+- BidirectionalTransformerBinarySelectionModel
+
+Example usage:
+    >>> import netam
+    >>> config = {'model_class': 'SingleValueBinarySelectionModel',
+    ...           'hparams': {'model_type': 'dasm'}}
+    >>> model = netam.create_selection_model_from_dict(config, torch.device('cpu'))
+"""
+
+import torch
+import json
+import inspect
+from typing import Dict, Any, List
+from pathlib import Path
+
+
+# Model presets for common configurations
+PRESET_CONFIGS = {
+    "single_default": {
```

Comment:
Seems surprising to have this config file infrastructure but also have dicts here directly. Shall we just have a configs dir?

Reply:
Or perhaps model_configs?